### PR TITLE
17903 Upgrade to Vue-CLI v5 + upgrade ESLint, etc + updates for Node 20

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#17903

*Description of changes:*
- app version = 5.2.0
- updated Dockerfile
- updated CD script
- updated CI script
- updated PR script
- added ESLint ignore file
- updated ESLint config
- updated Vue CLI to v5
- updated misc other dependencies (eslint, TS, etc)
- auto-fixed some code (let -> const)
- overrode misc eslint problems
- auto-fixed template attributes
- fixed some property/method visibilities
- fixed App unit test (incomplete store)
- fixed type error (object not iterable)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

**Review by JazzarKarim — APPROVED** [review]

**Comment by severinbeauvais** [comment]
@pwei1018 Can you help me find why `npm run build` is failing, please?

**Comment by severinbeauvais** [comment]
> @pwei1018 Can you help me find why `npm run build` is failing, please?

I will try upgrading to Vue-CLI v5. If that doesn't work then maybe we should consider changing to Vite. Thoughts?

File: src/App.vue
Comment (severinbeauvais) [inline src/App.vue:11]: This, and most changes in the vue files below, was auto-fixed by eslint.

File: src/components/common/applicant-info-1.vue
Comment (severinbeauvais) [inline src/components/common/applicant-info-1.vue:126]: "on" is not used here (weird workaround per comment above) so I simply ignored the eslint problem here.

I will test this before merging.

File: src/components/common/request-details.vue
Comment (severinbeauvais) [inline src/components/common/request-details.vue:24]: Hmm, weird auto-fix.

I will test this before merging.

File: src/components/common/request-details.vue
Comment (severinbeauvais) [inline src/components/common/request-details.vue:42]: I will test this before merging.

File: src/components/new-request/business-fetch.vue
Comment (severinbeauvais) [inline src/components/new-request/business-fetch.vue:36]: TS complained about this so I moved it into a new file in enums folder.

File: src/components/new-request/business-lookup.vue
Comment (severinbeauvais) [inline src/components/new-request/business-lookup.vue:8]: ESLint changed this and then I fixed it manually as per [similar code here](https://github.com/bcgov/bcrs-shared-components/blob/14b5f1e234826a04fb619588acbc05f058b935f6/src/components/business-lookup/BusinessLookup.vue#L5).

I will test this before merging.

File: src/components/new-request/name-check/name-check-conflicts.vue
Comment (severinbeauvais) [inline src/components/new-request/name-check/name-check-conflicts.vue:210]: A few visibility fixes here (as I was looking at other problems in this file).

File: src/components/new-request/submit-request/reserve-submit.vue
Comment (severinbeauvais) [inline src/components/new-request/submit-request/reserve-submit.vue:134]: ESLint didn't like the previous "@ts-ignore" comments so I worked around it this way.

File: src/interfaces/models.ts
Comment (severinbeauvais) [inline src/interfaces/models.ts:19]: I moved this below the declaration for `IssueI`.

Ditto for other changes in here.

File: src/interfaces/new-request-interface.ts
Comment (severinbeauvais) [inline src/interfaces/new-request-interface.ts:7]: Auto-fix. Pretty smart!

File: src/store/actions.ts
Comment (severinbeauvais) [inline src/store/actions.ts:260]: ESLint complained about `requestTypeCd` needing to be a const, but `request_action_cd` cannot be const, so I broke up this declaration like this.

File: Makefile
Comment (severinbeauvais) [inline Makefile:28]: Same as Create UI, Edit UI and Filings UI.

File: src/App.vue
Comment (severinbeauvais) [inline src/App.vue:293]: To get rid of a console.log during unit tests.

File: src/mixins/update-user-mixin.ts
Comment (severinbeauvais) [inline src/mixins/update-user-mixin.ts:10]: To get rid of a multiple "console.info" during unit tests.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-710-uutlgp48.web.app

File: src/components/common/applicant-info-1.vue
Comment (severinbeauvais) [reply src/components/common/applicant-info-1.vue:126]: Works fine.

image

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-710-uutlgp48.web.app

File: src/components/new-request/business-fetch.vue
Comment (JazzarKarim) [reply src/components/new-request/business-fetch.vue:36]: Out of curiosity, what was TS complaining about exactly Sev? That enums have to be separate and not be defined in the components?

Comment (severinbeauvais) [reply src/components/new-request/business-fetch.vue:36]: It said they were "unused", even though I was clearly using them 🤷‍♂️

**Comment by severinbeauvais** [comment]
Update: so this builds cleanly and the unit tests are happy (the few we have), but I'm experiencing this error while testing:

image

Thus, don't merge this until I have it all sorted out.

File: tests/unit/App.spec.ts
Comment (severinbeauvais) [inline tests/unit/App.spec.ts:19]: This is the fix for "isRoleStaff is not defined" and other getter/action test issues.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-710-uutlgp48.web.app

File: src/mixins/payment-mixin.ts
Comment (severinbeauvais) [inline src/mixins/payment-mixin.ts:76]: This is the fix for "paymentFees is not iterable".

The root cause is that several Vue components pass `paymentFees` to FeeSummary.vue, which expects an array of fee objects. However, `fees` is [stored as an object, not an array](https://github.com/bcgov/namerequest/blob/c8e8b6722adca77bccaca97d18c934e8506a2045/src/modules/payment/store/index.ts#L36). So, this code change now provides "paymentFees" as an array of a single object, and everyone is happy.

Comment (severinbeauvais) [reply src/mixins/payment-mixin.ts:76]: I'm guessing that the previous version of Typescript handled an object as equivalent to an array of a single object, which is why this worked before.

Here it is working now:

image

File: src/components/common/request-details.vue
Comment (severinbeauvais) [reply src/components/common/request-details.vue]: New on the left, old on the right. Works fine.

image

File: src/components/common/request-details.vue
Comment (severinbeauvais) [reply src/components/common/request-details.vue]: Same as above. (Works fine.)

Note that `class="col"` is NOT the final code. This is an old snapshot.

File: src/components/new-request/business-lookup.vue
Comment (severinbeauvais) [reply src/components/new-request/business-lookup.vue:8]: I broke this. I'm working on a fix now.

Comment (severinbeauvais) [reply src/components/new-request/business-lookup.vue]: I reverted the code change and added an ESLint rule exception instead.

I tested this and the business lookup works as before.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-710-uutlgp48.web.app

File: src/components/new-request/business-fetch.vue
Comment (JazzarKarim) [reply src/components/new-request/business-fetch.vue:36]: Well that's weird, haha. All right Sev.

**Review by eve-git — APPROVED** [review]
Excellent!

**Pull request merged**
